contracts/src/interfaces/IHyperdriveMatchingEngine.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import { IMorpho } from "morpho-blue/src/interfaces/IMorpho.sol";
+import { IMorphoFlashLoanCallback } from "morpho-blue/src/interfaces/IMorphoCallbacks.sol";
+import { IHyperdrive } from "../interfaces/IHyperdrive.sol";
+
+/// @title IHyperdriveMatchingEngine
+/// @notice Interface for the Hyperdrive matching engine.
+interface IHyperdriveMatchingEngine is IMorphoFlashLoanCallback {
+    /// @notice Thrown when an order is already cancelled.
+    error AlreadyCancelled();
+
+    /// @notice Thrown when an order is already expired.
+    error AlreadyExpired();
+
+    /// @notice Thrown when the destination for the add or remove liquidity
+    ///         options isn't configured to this contract.
+    error InvalidDestination();
+
+    /// @notice Thrown when orders that don't cross are matched.
+    error InvalidMatch();
+
+    /// @notice Thrown when the order type doesn't match the expected type.
+    error InvalidOrderType();
+
+    /// @notice Thrown when an address that didn't create an order tries to
+    ///         cancel it.
+    error InvalidSender();
+
+    /// @notice Thrown when `asBase = false` is used. This implementation is
+    ///         opinionated to keep the implementation simple.
+    error InvalidSettlementAsset();
+
+    /// @notice Thrown when the signature for an order intent doesn't recover to
+    ///         the expected signer address.
+    error InvalidSignature();
+
+    /// @notice Thrown when the long and short orders don't refer to the same
+    ///         Hyperdrive instance.
+    error MismatchedHyperdrive();
+
+    /// @notice Emitted when orders are cancelled
+    event OrdersCancelled(address indexed trader, bytes32[] orderHashes);
```

Comment:
Why the array of order hashes as opposed to `longOrderHash` and `shortOrderHash`?

Reply:
I'm not sure what you mean. This includes the hashes of all of the orders that were cancelled. It could be 1, 2, or 100 orders, so I'm not sure how this would fit into a long and short order hash structure.

Reply:
Derp sorry this definitely doesn't make sense... wrote this before reading the rest of the PR and forgot to remove.